### PR TITLE
drivers: clock_control: stm32f0/f3: fix PREDIV configuration

### PR DESCRIPTION
**Pull request by tagunil**
Some STM32F0/F3 parts have PREDIV1 prescaler after PLLSRC input mux; others have PREDIV prescaler before the mux just for the HSE output. This PR aims to fix PREDIV1 support.

Tested on Nucleo-F303RE board.

### CODE REVIEW

File: drivers/clock_control/Kconfig.stm32f0_f3 [hunk 1]
```diff
@@ -7,23 +7,25 @@ if SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X
 
 config CLOCK_STM32_PLL_PREDIV
 	int "PREDIV Prescaler"
+	depends on CLOCK_STM32_SYSCLK_SRC_PLL
 	default 1
 	range 1 16
 	help
-	  PREDIV is PLLSCR clock signal prescaler, allowed values: 1 - 16.
+	  PREDIV is a PLL clock signal prescaler for the HSE output.
```
Comment (FRASTM) [inline drivers/clock_control/Kconfig.stm32f0_f3:14]: so why removing  "depends on CLOCK_STM32_PLL_SRC_HSE"

Comment (tagunil) [reply drivers/clock_control/Kconfig.stm32f0_f3:14]: I'm removing it for CLOCK_STM32_PLL_PREDIV1 where it shouldn't be. There was no such dependency for CLOCK_STM32_PLL_PREDIV, and I'm not sure I should introduce it now.

Comment (FRASTM) [reply drivers/clock_control/Kconfig.stm32f0_f3:14]: ok

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
BK was stale

**Review by MaureenHelm — APPROVED** [review]

**Comment by nashif** [comment]
> BK was stale

this needs to be rebased!!

**Pull request merged**